src/io/calimero/dptxlator/DPTXlatorDateTime.java

```diff
@@ -982,7 +982,7 @@ else if (++day == 1) {
 						int dow = DAYS.length - 1;
 						while (dow >= 0 && !DAYS[dow].toLowerCase().startsWith(prefix))
 							--dow;
-						final boolean anyday = dow == 0 && t.hasMoreTokens() && t.nextToken().equalsIgnoreCase("day");
+						final boolean anyday = dow == 0 && t.hasMoreTokens() && "day".equalsIgnoreCase(t.nextToken());
```

Comment:
Switching the operands in cases like these seems unnecessary?

Reply:
Using ```"constant".eqals(somestringvariable)``` has the benefit that the string constant is never ```null``` and running ```equals()``` cannot cause a NPE. It cannot harm to write it in this order.

Reply:
That's true; the thing is, if `nextToken` returns null something is really really wrong. So a reversal like this basically hardens against the langspec/compiler/interpreter.